Change picker to Finder for import when running on macOS

johuether opened the pull request:
When running on macOS as an emulated iOS app we can use the Finder for picking image.

Till now the user could only chose from images in Apple Photos.

Now the user can chose from the whole filesystem. (But only files with image extension)

johuether committed:
Workaround for Android 14 image picker stripping EXIF information

- Do not use the new picker because it strips EXIF information
- Do not request Storage permission, because its unnecessary with file Intent on newer Android versions

johuether committed:
extend workaround for image picker to android 13

johuether committed:
Add Exif handling for HEIC images

johuether committed:
add picker for Finder when running on macOS

johuether committed:
Merge branch 'DavidWiesner:rma-androidx-33' into rma-androidx-33

sebastian-student commented on src/ios/CDVCamera.m:
```diff
@@ -459,56 +470,99 @@ - (NSData*)processImage:(UIImage*)image info:(NSDictionary*)info options:(CDVPic
-- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+- (void)documentPicker:(UIDocumentPickerViewController *)controller
+didPickDocumentsAtURLs:(NSArray<NSURL *> *)urls; {
     __weak CDVCamera* weakSelf = self;
```
This isn't used anywhere.

johuether replied:
Its used. It overrides an method. I dont know why XCode can not find the reference.

sebastian-student commented on src/ios/CDVCamera.m:
```diff
@@ -459,56 +470,99 @@ - (NSData*)processImage:(UIImage*)image info:(NSDictionary*)info options:(CDVPic
+    [self sendErrorResultWithMessage];
+}
+
+- (void)sendErrorResultWithMessage {
```
This does more than the naming implies. More like onEndSelectionWithoutImage. And use it also in other instances (e.g. insisde picker:didFinishPicking:)

johuether replied:
done. thanks

sebastian-student commented on src/ios/CDVCamera.m:
```diff
@@ -459,56 +470,99 @@ - (NSData*)processImage:(UIImage*)image info:(NSDictionary*)info options:(CDVPic
+
+- (void)handleImageFromPicker:(UIImage *)original withData:(NSData *)data {
+    
+    __weak CDVCamera* weakSelf = self;
```
Weak reference is unncessary at this point.

johuether replied:
done. thanks

sebastian-student reviewed (commented):
Don't see any critical problems.

johuether committed:
fix minor issues with macos picker

- remove unnecessary weak references
- change name of method to be more precise
